src/CachePlugin.php

```diff
@@ -34,31 +35,86 @@
-     *              we have to store the cache for a longer time than the server originally says it is valid for.
-     *              We store a cache item for $cache_lifetime + max age of the response.
-     *     @var array $methods list of request methods which can be cached.
-     * }
+     * @var bool $respect_cache_headers Whether to look at the cache directives or ignore them. This option is
+     *           deprecated, use `respect_response_cache_directives` instead
+     * @var int $default_ttl (seconds) If we do not respect cache headers or can't calculate a good ttl, use this
+     *           value
+     * @var string $hash_algo The hashing algorithm to use when generating cache keys
+     * @var int $cache_lifetime (seconds) To support serving a previous stale response when the server answers 304
+     *           we have to store the cache for a longer time than the server originally says it is valid for.
+     *           We store a cache item for $cache_lifetime + max age of the response.
+     * @var array $methods list of request methods which can be cached
+     * @var array $respect_response_cache_directives list of cache directives this plugin will respect while caching responses.
+     *           }.
      */
     public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
     {
         $this->pool = $pool;
         $this->streamFactory = $streamFactory;
 
+        if (isset($config['respect_cache_headers']) && $config['respect_response_cache_directives']) {
+            throw new \InvalidArgumentException(
+                'You can\'t provide config option "respect_cache_headers" and "respect_response_cache_directives". '.
+                'Use "respect_response_cache_directives" instead.'
+            );
+        }
+
         $optionsResolver = new OptionsResolver();
         $this->configureOptions($optionsResolver);
         $this->config = $optionsResolver->resolve($config);
     }
 
     /**
+     * This method will setup the cachePlugin in client cache mode. When using the client cache mode the plugin will
+     * cache responses with `private` cache directive.
+     *
+     * @param CacheItemPoolInterface $pool
+     * @param StreamFactory          $streamFactory
+     * @param array                  $config        For all possible config options see the constructor docs
+     *
+     * @return CachePlugin
+     */
+    public static function clientCache(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
+    {
+        // Allow caching of private requests
+        if (isset($config['respect_response_cache_directives'])) {
+            $config['respect_response_cache_directives'][] = 'no-cache';
+            $config['respect_response_cache_directives'][] = 'max-age';
+            array_unique($config['respect_response_cache_directives']);
```

Comment:
afaik array_unique does not modify the input array. you need to re-assign the value to the field.